libraries/extractor/src/main/java/androidx/media3/extractor/DtsUtil.java

```diff
@@ -47,55 +109,144 @@ public final class DtsUtil {
+  /** Maps AMODE to the number of channels. See ETSI TS 102 114 table 5-4. */
   private static final int[] CHANNELS_BY_AMODE =
       new int[] {1, 2, 2, 2, 2, 3, 3, 4, 4, 5, 6, 6, 6, 7, 8, 8};
 
-  /** Maps SFREQ to the sampling frequency in Hz. See ETSI TS 102 144 table 5.5. */
+  /** Maps SFREQ to the sampling frequency in Hz. See ETSI TS 102 114 table 5-5. */
   private static final int[] SAMPLE_RATE_BY_SFREQ =
       new int[] {
-        -1, 8000, 16000, 32000, -1, -1, 11025, 22050, 44100, -1, -1, 12000, 24000, 48000, -1, -1
+          -1, 8_000, 16_000, 32_000, -1, -1, 11_025, 22_050, 44_100, -1, -1, 12_000, 24_000, 48_000,
+          -1, -1
       };
 
-  /** Maps RATE to 2 * bitrate in kbit/s. See ETSI TS 102 144 table 5.7. */
+  /** Maps RATE to 2 * bitrate in kbit/s. See ETSI TS 102 114 table 5-7. */
   private static final int[] TWICE_BITRATE_KBPS_BY_RATE =
       new int[] {
-        64, 112, 128, 192, 224, 256, 384, 448, 512, 640, 768, 896, 1024, 1152, 1280, 1536, 1920,
-        2048, 2304, 2560, 2688, 2816, 2823, 2944, 3072, 3840, 4096, 6144, 7680
+          64, 112, 128, 192, 224, 256, 384, 448, 512, 640, 768, 896, 1_024, 1_152, 1_280, 1_536,
+          1_920, 2_048, 2_304, 2_560, 2_688, 2_816, 2_823, 2_944, 3_072, 3_840, 4_096, 6_144, 7_680
+      };
+
+  /**
+   * Maps MaxSampleRate index to sampling frequency in Hz. See ETSI TS 102 114 V1.6.1 (2019-08)
+   * table 7-9.
+   */
+  private static final int[] SAMPLE_RATE_BY_INDEX =
+      new int[] {
+          8_000, 16_000, 32_000, 64_000, 128_000, 22_050, 44_100, 88_200, 176_400, 352_800, 12_000,
+          24_000, 48_000, 96_000, 192_000, 384_000
       };
 
   /**
-   * Returns whether a given integer matches a DTS sync word. Synchronization and storage modes are
-   * defined in ETSI TS 102 114 V1.1.1 (2002-08), Section 5.3.
+   * Maps RefClockCode (Reference Clock Code) to reference clock period. See ETSI TS 102 114 V1.6.1
+   * (2019-08) table 7-3.
+   */
+  private static final int[] REF_CLOCK_FREQUENCY_BY_CODE = new int[] {32_000, 44_100, 48_000};
+
+  /**
+   * Index look-up table corresponding to the prefix code read from the bitstream. See ETSI TS 103
+   * 491 V1.2.1 (2019-05) Table 5-2: ExtractVarLenBitFields.
+   */
+  private static final int[] VAR_INT_INDEX_TABLE = new int[] {0, 0, 0, 0, 1, 1, 2, 3};
+
+  /**
+   * Number of bits used to encode a field. See ETSI TS 103 491 V1.2.1 (2019-05) Table 5-2:
+   * ExtractVarLenBitFields.
+   */
+  private static final int[] VAR_INT_BITS_USED = new int[] {1, 1, 1, 1, 2, 2, 3, 3};
+
+  /**
+   * Maps index to number of clock cycles in the bfase duration period. See ETSI TS 103 491 V1.2.1
+   * (2019-05) Table 6-13.
+   */
+  private static final int[] BASE_DURATION_BY_INDEX = new int[] {512, 480, 384};
+
+  /**
+   * A state variable that stores the UHD audio chunk ID extracted from the FTOC sync frame.
+   * This value will be used in the subsequent FTOC non-sync frame */
+  private static int storedUhdAudioChunkId;
```

Comment:
I don't think this can be static.

What if we have two ExoPlayer instances in the same app, both playing UHD content? They will end up cross-talking in this field.

Reply:
@icbaker Would you think that the two instance executions will be affected by the other static variables in the Utils class(unless the instance switches the execution in the frame boundary)?
What would be a use case for having two instances playing UHD content?

In my latest commit([39cd88d](https://github.com/androidx/media/pull/275/commits/39cd88d2d8d7360c42d4927ee6b3f32e96b9b91b), I have updated the logic to allocate the state variable in the DtsReader class instead of the DtsUtil class. I hope that this will ensure an independent state for each of the ExoPlayer instances.

Reply:
> Would you think that the two instance executions will be affected by the other static variables in the Utils class

Which static variables in `DtsUtil` are you referring to? All the ones I can see are either immutable primitives, or arrays that aren't technically immutable but aren't meant to be mutated (and if they're being mutated, that's probably a bug we should fix).

> What would be a use case for having two instances playing UHD content?

One example might be cross-fading between two tracks, which means playing both at the same time for a bit. Or otherwise mixing content together. Lots of reasons we shouldn't assume there's only ever one ExoPlayer instance per app and therefore generally avoid static state.

----

The proposed approach of keeping track of the state in the reader looks better - thanks.

Reply:
Thank you for explaining the details.
I hope that all the review comments are addressed and no further modification is required on this CL.

Reply:
Hi @icbaker , @rohitjoins 
Would you think this CL is ready to merge or need any updates?
Thank you.

Reply:
We expect to see this in 1.2.0.

Reply:
Hi @rahulnmohan,

This is still going through the internal review process. We don't expect this to be released as part of 1.2.0 as the changes are big enough, it cannot be included after 1.2.0-alpha01 release which was done more than 3 months ago.

Reply:
@rohitjoins @FongMi We have several customers in China who are using DTS audio with HLS-TS and Exoplayer. This contribution will go a long way in helping us support these efforts. Your kind assistance to expedite this is very much appreciated. Thanks.

Reply:
@cedricxperi Can you help me test this link?
http://150.158.170.214/ishang.php?id=cctv1hd
need china vpn, I recommand 穿梭, and use 全局模式
EXO don't decode the video track, only audio track
IJK can play it.

Reply:
@FongMi I tested with Transocks VPN. It can't play video or audio. I am testing from Singapore. May I know what is the content?

Reply:
Sorry, it not about this pull, I have mail to you.